task.go

```diff
@@ -31,9 +32,13 @@ type Log struct {
 	Stdout string `json:"stdout"`
 }
 
-// State returns inforation about rsync processing task
-func (t Task) State() State {
-	return *t.state
+// State returns information about rsync processing task
+// lock mutex to avoid accessing it while ProcessStdout is writing to it
+func (t *Task) State() State {
+	t.mutex.Lock()
+	c := *t.state
+	t.mutex.Unlock()
+	return c
 }
 
 // Log return structure which contains raw stderr and stdout outputs
```

Comment:
It doesn't affect us, but just for completeness, I think we should probably use the mutex in `Task.Log()` method as you did in `Task.State()`, just in case there is someone trying to read logs in real time.

Reply:
yep, you are rigth, a test for this triggers a data race, updating to also lock that part